classes/export_webservices.php

```diff
@@ -0,0 +1,387 @@
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_wsformat;
+
+use core_external\external_api;
+
+use core_external\external_multiple_structure;
+use core_external\external_single_structure;
+
+/**
+ * Class for processing and exporting web service data.
+ */
+class export_webservices {
+
+    /**
+     * Stores the host address of the server.
+     *
+     * @var string
+     */
+    private $host = '';
+
+    /**
+     * An array to hold webservice objects.
+     *
+     * @var array
+     */
+    public $webservices = [];
+
+
+    /**
+     * Constructor function - assign instance variables.
+     *
+     * @param string $host
+     * @param array  $selectedwebserviceindices
+     */
+    public function __construct(string $host, array $selectedwebserviceindices) {
+        $this->host        = $host;
+        $this->webservices = $this->get_selected_webservice_objects($selectedwebserviceindices);
+    }
+
+    /**
+     * Exports data as cURL commands in a text file.
+     * Sets header to initiate download with filename and extension.
+     */
+    public function export_as_curl(): void {
+        header('Content-Disposition: attachment; filename=curl.txt');
+        header('Content-Type: application/plain');
+
+        $curlstrings = [];
+        foreach ($this->webservices as $webservice) {
+            $paramsarray = $this->get_formatted_param_array($webservice);
+
+            $curlstrings[] = 'curl ' . $this->create_request_string($webservice, $paramsarray);
+        }
+
+        foreach ($curlstrings as $curlstring) {
+            echo $curlstring . "\n" . "\n";
+        }
+    }
+
+
+    /**
+     * Exports data as Postman Collection in a json file.
+     * Sets header to initiate download with filename and extension.
+     */
+    public function export_as_postman(): void {
+        header('Content-Disposition: attachment; filename=postman.json');
+        header('Content-Type: application/json');
+
+        $postmanitems = [];
+        foreach ($this->webservices as $webservice) {
+            $paramsarray = $this->get_formatted_param_array($webservice);
+
+            $postmanitems[] = $this->create_postman_request_item($webservice, $paramsarray);
+        }
+
+        $postmancollection = $this->create_postman_collection($postmanitems);
+        $beautifiedjson    = json_encode($postmancollection, JSON_PRETTY_PRINT);
+        echo $beautifiedjson;
+    }
+
+
+    /**
+     * Retrieves an array of webservice objects based on provided indices.
+     *
+     * @param  array $selectedwebserviceindices An array of indices to grab the webservice objects.
+     * @return array An array of webservice objects.
+     */
+    private function get_selected_webservice_objects(array $selectedwebserviceindices): array {
+        $webservicesrecords = $this->get_indexed_webservice_records();
+
+        $webservices = [];
+        foreach ($selectedwebserviceindices as $index) {
```

Comment:
This is fine as is, but it is possible in PHP to use arrays as a map, and so you could have [1 => 'a'] or even ['a' => 'b'], and you can address these individually in a for loop such as `for ($array as $key => $value)` and use each inside of the block. Just noting